JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,329 @@
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET client_id= '" + CLIENT_ID + "';");
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET client_secret= '" + CLIENT_SECRET + "';");
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET introspect_url = '" + INTROSPECT_URL + "';");
+		st.executeUpdate("CREATE USER " + USER + ";");
+		st.executeUpdate("GRANT AUTHENTICATION v_oauth TO " + USER + ";");
+		st.executeUpdate("GRANT ALL ON SCHEMA PUBLIC TO " + USER +";");
+		st.close();
+	}
+	// Dispose the authentication record from database
+	private static void TearDown() 
+	{
+		try
+		{
+			Statement st = vConnection.createStatement(); 
+			String USER = prop.getProperty("User");
+			st.executeUpdate("DROP USER IF EXISTS " + USER + " CASCADE");
+			st.executeUpdate("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE");
+			vConnection.close();
+		}catch(Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+	// Connect to Database using access Token
+	private static Connection connectToDB(String accessToken) throws SQLException 
+	{
+		Properties jdbcOptions = new Properties();
+		jdbcOptions.put("oauthaccesstoken", accessToken);
+		return DriverManager.getConnection(
+				"jdbc:vertica://" + getParam("Host") + ":" + getParam("Port") + "/" + getParam("Database") , jdbcOptions);
     }
-
-    private static ResultSet executeQuery(Connection conn) throws SQLException {
-        Statement stmt = conn.createStatement();
-        return stmt.executeQuery("SELECT user_id, user_name FROM USERS ORDER BY user_id");
+	// Test connection using access token and test database query result
+	private static void ConnectToDatabase() throws SQLException 
+	{
+		int connAttemptCount = 0;
+		while (connAttemptCount <= 1)
+        {
+            try
+            {
+            	String accessToken = System.getProperty(OAUTH_ACCESS_TOKEN_VAR_STRING);
+        		if( null == accessToken || accessToken.isEmpty())
+        		{
+					throw new Exception("Access Token is not available.");
+        		}else
+        		{
+        			Connection conn = connectToDB(accessToken);
+        			ResultSet rs = executeQuery(conn);
+        			printResults(rs);
+					break;
+        		}
+            }catch(Exception ex) 
+            {
+            	if (connAttemptCount > 0) 
+            	{ 
+            		break;
+            	}
+            	try {
+					ex.printStackTrace();
+            		GetTokensByRefreshGrant();
+            	}catch (Exception e1)
+                {
+            		e1.printStackTrace();
+                    try
+					{
+						GetTokensByPasswordGrant();
+					}catch(Exception e2)
+					{
+						e2.printStackTrace();
+					}
+                }
+            	++connAttemptCount;
+            }
+        }
+	}
+	// execute Simple query on database connection
+	private static ResultSet executeQuery(Connection conn) throws SQLException 
+	{
+		Statement stmt = conn.createStatement();
+		return stmt.executeQuery("SELECT user_id, user_name FROM USERS ORDER BY user_id");
```

Comment:
We should print something like "Connected Successfully. The current user is: " + the query result.

Reply:
Here all the users available in the USERS table are getting printed from database table USERS. Added print to specify "executing query :" + query